pkg/server/endpoints/node/handler.go

```diff
@@ -1085,15 +1095,16 @@ func (h *Handler) parseCSR(csrBytes []byte, mode idutil.ValidationMode) (*CSR, e
 		return nil, errorutil.WrapError(err, "failed to parse CSR")
 	}
 
-	var spiffeID string
+	var spiffeID spiffeid.ID
 	switch len(csr.URIs) {
 	case 0:
 	case 1:
-		spiffeID, err = idutil.NormalizeSpiffeID(csr.URIs[0].String(), mode)
+		id, err := idutil.NormalizeSpiffeID(csr.URIs[0].String(), mode)
 		if err != nil {
 			h.c.Log.WithError(err).Error("Invalid SPIFFE ID in CSR")
 			return nil, errorutil.WrapError(err, "invalid SPIFFE ID in CSR")
 		}
+		spiffeID = spiffeid.RequireFromString(id)
```

Comment:
In practice this is probably ok since the code before should be returning a well-formed ID but in general I don't think we should be calling methods that can panic from the production code. I know it's a little more verbose but until the idutil validation helpers all use `spiffeid.ID`, we should probably use `spiffeid.FromString` and just check for an error in this production code.